tests/rbd/rbd_tier0.py

```diff
@@ -0,0 +1,299 @@
+                  clone: rbd_rep_pool_clone
+               operations:
+                  map: true
+                  io: true
+                  nounmap: false
+
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+    """
+    try:
+        pool = kw["config"][pool_type]["pool"]
+        image = kw["config"][pool_type]["image"]
+        size = kw["config"][pool_type]["size"]
+        size_int = int(re.findall(r"\d+", size)[0])
+        snap = kw["config"][pool_type].get("snap", f"{image}_snap")
+        clone = kw["config"][pool_type].get("clone", f"{image}_clone")
+        image_spec = f"{pool}/{image}"
+        snap_name = f"{pool}/{image}@{snap}"
+
+        # Run fio and krbd map operations on image
+        kw.get("config")["image_spec"] = [image_spec]
+        kw.get("config")["size"] = size
+        kw["rbd_obj"] = rbd
+        krbd_io_handler(**kw)
+
+        # List images in a pool and verify
+        image_list = rbd.list_images(pool)
+        if image not in image_list:
+            log.error(f"Image {image} in not present in images listed for pool {pool}")
+            return 1
+
+        # Display image info and verify
+        image_info = rbd.image_info(pool, image)
+        log.info(f"Image info : {image_info}")
+        if image not in image_info["name"]:
+            log.error(f"Image {image} in not listed for pool {pool}")
+            return 1
+
+        # List image metadata, set last_update metadata and remove it and verify
+        image_meta = rbd.image_meta(action="list", image_spec=image_spec)
+        try:
+            image_meta = json.loads(image_meta)
+            log.info(f"Image meta : {image_meta}")
+        except ValueError as e:
+            log.error(f"Image meta fetch failed : {image_meta} with error {e}")
+
+        # verify image info and log any errors
+
+        # Set last_update metadata and remove it and verify both
+        date = datetime.today().strftime("%Y-%m-%d")
+        rbd.image_meta(
+            action="set",
+            image_spec=image_spec,
+            key="last_update",
+            value=date,
+        )
+
+        update_date = rbd.image_meta(
+            action="get", image_spec=image_spec, key="last_update"
+        )
+        log.info(f"Updated date : {update_date}")
+        if update_date.strip() != date:
+            log.error("Setting image metadata last_update failed")
+            return 1
+
+        rbd.image_meta(
+            action="remove",
+            image_spec=image_spec,
+            key="last_update",
+        )
+        update_date = rbd.image_meta(
+            action="get", image_spec=image_spec, key="last_update"
+        )
+        log.info(f"Updated date : {update_date}")
+        if update_date != 1:
+            log.error("Removing image metadata last_update failed")
+            return 1
+
+        # Create snapshot, protect, clone, flatten, resize, remove, rollback
+        if rbd.snap_create(pool, image, snap) != 0:
```

Comment:
I think we can avoid `!=0` when just` if rbd.snap_create(pool, image, snap): ` does the same?
also same to other places used in that way.

Reply:
Done.

Reply:
Done.